06-classification_continued.Rmd

```diff
@@ -194,18 +197,100 @@ cancer_test_predictions %>%
   metrics(truth = Class, estimate = .pred_class)
 ```
 
-This shows that the accuracy of the classifier on the test data was 88%.
+```{r 06-accuracy-2, echo = FALSE, warning = FALSE}
+cancer_acc_1 <- cancer_test_predictions %>% 
+                metrics(truth = Class, estimate = .pred_class) %>% 
+                filter(.metric == 'accuracy')
+```
+
+This shows that the accuracy of the classifier on the test data 
+was `r round(100*cancer_acc_1$.estimate, 0)`%.
 We can also look at the *confusion matrix* for the classifier, which shows
 the table of predicted labels and correct labels, using the `conf_mat` function:
 
 ```{r 06-confusionmat}
-cancer_test_predictions %>%
-  conf_mat(truth = Class, estimate = .pred_class)
+confusion <- cancer_test_predictions %>%
+             conf_mat(truth = Class, estimate = .pred_class)
+confusion
+```
+
+```{r 06-confusionmat-2, echo = FALSE}
+confusionmt <- confusion %>% tidy()
+confu11 <- (confusionmt %>% filter(name == "cell_1_1"))$value
+confu12 <- (confusionmt %>% filter(name == "cell_1_2"))$value
+confu21 <- (confusionmt %>% filter(name == "cell_2_1"))$value
+confu22 <- (confusionmt %>% filter(name == "cell_2_2"))$value
+```
+
+This table shows that the classifier labelled 
+`r confu11`+`r confu22` = `r confu11+confu22` observations 
+correctly, `r confu21` observations as benign when they were truly malignant,
+and `r confu12` observations as malignant when they were truly benign. 
```

Comment:
Maybe also add a sentence here as why we want to look at this confusion matrix, like why not just look at the overall accuracy? What can this tell us that overall accuracy cannot. UPDATE: I see this below now. Still do the text change I suggest below to make the sentence more digestible to learners.
```suggestion
This table shows that the classifier labelled 
`r confu11`+`r confu22` = `r confu11+confu22` observations 
correctly. It also shows us that it made some mistakes, it labelled `r confu21` observations as benign when they were truly malignant,
and `r confu12` observations as malignant when they were truly benign.  
```

Reply:
done